### DISCUSSION TITLE
Can a ram disk be used instead of the SSD to run the benchmark and permutor-cli tool?

### DISCUSSION THREAD

**Discussion by e-dong** [Proryanator/encoder-benchmark#24] [Q&A] [answered]
I realized my SSD's sequential reads is not fast enough to keep up with >= 2k@120. Is it possible to load the file into RAM instead? 

According to this, the RAM speed is much faster than disk.
https://www.crucial.com/support/memory-speeds-compatability

The 4k-120 file is 28 G, so I could use 32 GB at a minimum. But with only 4 GB of free ram, that could impact performance?

**Comment by Proryanator** [comment, marked as answer]
Hey there! It could be possible to do it all in RAM. I considered adding a RAM mode to get around SSD/storage limitations initially but never put work to it.

Either loading the whole video file at once into RAM or doing some clever partial load && read from your storage to keep your memory packed.

Would you be interested in a RAM mode of this tool? 😊

**Reply by e-dong** [reply]
Yeah definitely! I don't really need a new SSD, so I would prefer a RAM mode / approach. I only have 16 GB, but would definitely prefer to get more RAM than a  new SSD haha.

These are my current specs:
```
CPU: Intel - Core i7-8700K 3.7GHz 6-Core Processor
CPU Cooler: Cooler Master - Hyper 212 EVO 82.9 CFM Sleeve Bearing CPU CooMOBOler
MOBO: Asus - ROG MAXIMUS X HERO (WI-FI AC) ATX LGA1151 Motherboard
RAM: Corsair - Vengeance LPX 16GB (2 x 8GB) DDR4-2400 Memory
SSD: Samsung - 850 EVO-Series 500GB 2.5" Solid State Drive
HDD:Seagate - Barracuda 2TB 3.5" 7200RPM Internal Hard Drive
GRAPHIC CARD: RTK 3070
CHASSIS: Fractal Design - Define R6 White TG ATX Mid Tower Case
PSU: Corsair - 1000W 80+ Platinum Certified Fully-Modular ATX Power Supply
```

**Reply by Proryanator** [reply]
https://github.com/Proryanator/encoder-benchmark/issues/25

I added an issue for this! Would be fun to work on I think, and definitely would make this tool more accessible. (especially for < 2K video files).
